sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/NonStreamingOrderByUtils.java

```diff
@@ -0,0 +1,114 @@
+            .toArray(Flux[]::new);
+        return Flux.mergeComparingDelayError(1,consumeComparer, fluxes);
+    }
+
+    private static Flux<OrderByRowResult<Document>> toNonStreamingOrderByQueryResultObservable(DocumentProducer<Document> producer,
+                                                                                               RequestChargeTracker tracker,
+                                                                                               Map<String, QueryMetrics> queryMetricsMap,
+                                                                                               int initialPageSize,
+                                                                                               OrderbyRowComparer<Document> consumeComparer,
+                                                                                               Collection<ClientSideRequestStatistics> clientSideRequestStatisticsList) {
+        return producer
+            .produceAsync()
+            .transformDeferred(new NonStreamingOrderByUtils.PageToItemTransformer(tracker, queryMetricsMap, initialPageSize,
+                consumeComparer, clientSideRequestStatisticsList));
+    }
+
+    private static class PageToItemTransformer implements
+        Function<Flux<DocumentProducer<Document>.DocumentProducerFeedResponse>, Flux<OrderByRowResult<Document>>> {
+        private final RequestChargeTracker tracker;
+        private final Map<String, QueryMetrics> queryMetricsMap;
+        private final Integer initialPageSize;
+        private final OrderbyRowComparer<Document> consumeComparer;
+        private final Collection<ClientSideRequestStatistics> clientSideRequestStatistics;
+
+        private PageToItemTransformer(RequestChargeTracker tracker, Map<String, QueryMetrics> queryMetricsMap,
+                                      Integer initialPageSize, OrderbyRowComparer<Document> consumeComparer,
+                                      Collection<ClientSideRequestStatistics> clientSideRequestStatistics) {
+            this.tracker = tracker;
+            this.queryMetricsMap = queryMetricsMap;
+            this.initialPageSize = initialPageSize;
+            this.consumeComparer = consumeComparer;
+            this.clientSideRequestStatistics = clientSideRequestStatistics;
+        }
+
+        @Override
+        public Flux<OrderByRowResult<Document>> apply(Flux<DocumentProducer<Document>.DocumentProducerFeedResponse> source) {
+            // the size of the priority queue is set to size+1, because when the pq reaches the max size we add that
+            // item and then remove the element. If we don't do this, then when adding this element the size of the pq
+            // will be increased automatically by 50% and then there would be inconsistent results for later pages.
+            PriorityBlockingQueue<OrderByRowResult<Document>> priorityQueue = new PriorityBlockingQueue<>(initialPageSize + 1, consumeComparer);
```

Comment:
A dumb question:
- The PriorityQueue here basically sorts on for each `FeedRange` basis and this is further sorted when you merge the results from each `FeedRange`?